contracts/liquidityMining/KyberFairLaunch.sol

```diff
@@ -402,35 +447,42 @@ contract KyberFairLaunch is IKyberFairLaunch, PermissionAdmin, ReentrancyGuard {
     uint256 _pid,
     bool shouldHarvest
   ) internal {
-    uint128 lastAccRewardPerShare = poolInfo[_pid].accRewardPerShare;
-    UserInfo memory user = userInfo[_pid][_to];
+    uint256 userAmount = userInfo[_pid][_to].amount;
+    address[] memory rTokens = poolInfo[_pid].rewardTokens;
 
-    if (user.amount == 0) {
+    if (userAmount == 0) {
       // update user last reward per share to the latest pool reward per share
       // by right if user.amount is 0, user.unclaimedReward should be 0 as well,
       // except when user uses emergencyWithdraw function
-      userInfo[_pid][_to].lastRewardPerShare = lastAccRewardPerShare;
+      for(uint256 i = 0; i < rTokens.length; i++) {
+        userInfo[_pid][_to].userRewardData[rTokens[i]].lastRewardPerShare =
+          poolInfo[_pid].poolRewardData[rTokens[i]].accRewardPerShare;
+      }
       return;
     }
 
-    // user's unclaim reward + user's amount * (pool's accRewardPerShare - user's lastRewardPerShare) / precision
-    uint256 _pending = user.amount.mul(
-      uint256(lastAccRewardPerShare).sub(user.lastRewardPerShare)
-    ) / PRECISION;
-    _pending = _pending.add(user.unclaimedReward);
-
-    if (shouldHarvest) {
-      userInfo[_pid][_to].unclaimedReward = 0;
-      if (_pending > 0) {
-        rewardLocker.lock(rewardToken, _to, _pending);
-        emit Harvest(_to, _pid, block.number, _pending);
+    for(uint256 i = 0; i < rTokens.length; i++) {
+      uint256 lastAccRewardPerShare = poolInfo[_pid].poolRewardData[rTokens[i]].accRewardPerShare;
+      UserRewardData storage rewardData = userInfo[_pid][_to].userRewardData[rTokens[i]];
+      // user's unclaim reward + user's amount * (pool's accRewardPerShare - user's lastRewardPerShare) / precision
+      uint256 _pending = userAmount.mul(
+        lastAccRewardPerShare.sub(rewardData.lastRewardPerShare)
+      ) / PRECISION;
+      _pending = _pending.add(rewardData.unclaimedReward);
+
+      if (shouldHarvest) {
```

Comment:
may be this is more clear
https://github.com/KyberNetwork/dao_sc/blob/liquidity_mining_multiple_reward/contracts/liquidityMining/KyberFairLaunchv2.sol#L467-L472